src/slack.js

```diff
@@ -0,0 +1,65 @@
+const fs = require('fs');
+const AWS = require('aws-sdk');
+const Slack = require('slack-node');
+
+const logger = require('./logger');
+
+const instance = new Slack(process.env.ACCESS_TOKEN);
+
+const uploadToS3 = async (key, metadata, tmpPath) => {
+  const s3 = new AWS.S3();
+
+  try {
+    return s3
+      .upload({
+        Bucket: process.env.S3_BUCKET,
+        Key: `emoji/key`,
+        // Body: writeStream,
+        Body: fs.createReadStream(tmpPath),
+        ContentType: metadata.file.mimetype,
+        ACL: 'public-read',
+      })
+      .promise();
+  } catch (e) {
+    logger.error(`Failed to upload to s3: ${tmpPath} %s`, e);
+  }
+};
+
+const slack = (method, params) => {
+  return new Promise((resolve, reject) => {
+    instance.api(method, params, (err, response) => {
+      err || !response.ok
+        ? reject(err || new Error(`Response from slack ${response.error}`))
+        : resolve(response);
+    });
+  });
+};
+
+// Now that we're checking the file upload has an alias, we could probably revert
+// this back to actually uploading a file to the channel rather than sending a
+// message with an attachment which would mean we didn't need to upload to s3.
+// I haven't removed it in the event that we're keeping all of them for a showcase
+// website which we discussed briefly.
```

Comment:
I think we will still want to send it back for the user confirmation before it actually gets uploaded to slack/github?

Reply:
yeah but what i mean is we can directly upload the file to slack as an "upload" (on `files.upload` iirc) like we had previously but i was getting a loop because it would ping the hook, resize the file, upload the file which would trigger the hook again and get stuck infinitely.

The above won't happen any more because the upload caused by emojo won't have an :alias: message so it won't loop any more.

This would remove the need for the s3 bucket and waiting to upload it etc.